x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/ShardFollowTaskReplicationTests.java

```diff
@@ -159,9 +159,19 @@ private ReplicationGroup createFollowGroup(int replicas) throws IOException {
     }
 
     private ShardFollowNodeTask createShardFollowTask(ReplicationGroup leaderGroup, ReplicationGroup followerGroup) {
-        ShardFollowTask params = new ShardFollowTask(null, new ShardId("follow_index", "", 0),
-            new ShardId("leader_index", "", 0), between(1, 64), between(1, 8), Long.MAX_VALUE, between(1, 4), 10240,
-            TimeValue.timeValueMillis(10), TimeValue.timeValueMillis(10), Collections.emptyMap());
+        ShardFollowTask params = new ShardFollowTask(
```

Comment:
I have added a test here: https://github.com/elastic/elasticsearch/pull/33546/commits/9620d7bf1a40dd177d316f896b0b397be4a69bef

I added a method to `ESIndexLevelReplicationTestCase` so that I can reinit the primary shard. Which then allowed me to change the history uuid and restart the shard in the test. I think this an okay test approach, what do you think?

Reply:
I am ok with this approach. How about just changing a historyUUID in the requests of the follow-task without really bootstrapping new history on the leader?

Reply:
> How about just changing a historyUUID in the requests of the follow-task without really bootstrapping new history on the leader?

That would change the recorded history uuid, which is immutable in non test scenarios. It would trigger the same error to be thrown, but it feels reverse to me. In this test now the current history uuid changes, which I think is more realistic.